### PR TITLE
More reliable ec_canvas tests.

### PR DESCRIPTION
**Pull request by islemaster**
Instead of waiting an arbitrary time and failing if things happen
to not be ready yet, defer until a setup predicate is met, leaning
on the built-in test timeout (17 seconds right now) to avoid
infinite wait.

Full suite passed 5/5 times, which I couldn't do prior to this change.

### CODE REVIEW

File: apps/test/solutions/applab/ec_canvas.js [hunk 1]
```diff
@@ -191,3 +185,12 @@ module.exports = {
     }
   ]
 };
+
+var canvasAndResultsPopulated = function () {
```
Comment (Bjvanminnen) [inline apps/test/solutions/applab/ec_canvas.js]: I think our style guide says top level functions should be in the form

```
function canvasAndResultsPopulate...
```

Though I don't care too much in this case.

File: apps/test/util/testUtils.js [hunk 2]
```diff
@@ -132,6 +133,40 @@ exports.runOnAppTick = function (app, tick, fn) {
...
+ * @example
+ *   tickAppUntil(Applab, function () {
+ *     return document.getElementById('slow-element');
+ *   }).then(function () {
+ *     assert(document.getElementById('slow-element').textContent === 'pass');
+ *   });
+ */
+exports.tickAppUntil = function (app, predicate) {
+  if (!app) {
+    throw new Error('not supported outside of studio/applab');
```
Comment (Bjvanminnen) [inline apps/test/util/testUtils.js]: This conditional just checks that I pass something, not that this something is studio/applab. I wonder if the check should be

```
if (!app || !app.onTick)
```

**Comment by Bjvanminnen** [comment]
couple nits, otherwise lgtm

File: apps/test/util/testUtils.js (hunk 2 above)
Comment (islemaster) [inline apps/test/util/testUtils.js]: :+1: Sure, I'm pro-ducktyping here.

**Pull request merged**